### PR TITLE
Fixes some more holes in the ref tracker

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

Gives the reference tracker the ability to catch the harddels related to #58945 
The reference tracker was failing to check null keyed assoc list entries, along with being unable to check both lists in a list(list() = list()) pair
This resolves that, and adds some new logic to the unit test to check for this sort of thing

## Why It's Good For The Game

The more concrete this thing is, the more I can force people to use it, and the less harddels there'll be

### CODE REVIEW

File: code/modules/unit_tests/find_reference_sanity.dm [hunk 1]
```diff
@@ -67,3 +67,25 @@
...
+	//Calm before the storm
+	testbed.test_assoc_list = list(null = victim)
+
+	victim.DoSearchVar(testbed, "Fourth Run", search_time = 5)
+	TEST_ASSERT(testbed.test_assoc_list, "The ref-tracking tool failed to find a null key'd assoc list entry")
+	victim.found_refs.Cut()
+	testbed.test_assoc_list.Cut()
+
+	//Let's do some more complex assoc list investigation
+	var/list/to_find_in_key = list(victim)
```
Comment (Mothblocks) [inline code/modules/unit_tests/find_reference_sanity.dm:80]: Make this a separate test.

Comment (LemonInTheDark) [reply code/modules/unit_tests/find_reference_sanity.dm:80]: why? it's the same concept as the rest of the unit test, several different testcases all of which test the stability of DoSearchVar(). Is it a bloat thing?

Comment (Mothblocks) [reply code/modules/unit_tests/find_reference_sanity.dm:80]: Unit tests should, ideally, be testing one behavior each.

Comment (LemonInTheDark) [reply code/modules/unit_tests/find_reference_sanity.dm:80]: Should I give each one its own file? or keep them in the same place but under different headers

Comment (LemonInTheDark) [reply code/modules/unit_tests/find_reference_sanity.dm:81]: I split them, will this do?

**Pull request merged**